### PR TITLE
Issue 28: Add initial doc page for listening and tracing

### PR DESCRIPTION
**Pull request by WilliamParker**

### CODE REVIEW

**Comment by WilliamParker** [comment]
I'm not sure if anyone will want to review this; if nobody speaks up after a few days I'll probably merge this.  The current state is that we no documentation, so having something even if we'll iterate on it later seems like clear progress.

**Review by mrrodriguez — APPROVED** [review]
I think it is good as it is as a initial doc. I read through it and didn't see anything that seemed incorrect etc.

**Comment by Zylox** [comment]
Looks like whatever we are hosting on uses the language exstensions to code blocks, so you should probably add `clj` after the first "```"
Examples of the difference it makes:
#### before
screen shot 2018-03-06 at 10 24 15 am
#### after
screen shot 2018-03-06 at 10 24 28 am

File: _docs/listeners.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+(-> (with-tracing (mk-session [too-cold-rule]))
+    (insert (->Temperature -60 "Alaska"))
+    fire-rules
+    (retract (->Temperature -60 "Alaska"))
+    fire-rules
+    (insert (->Temperature -70 "Alaska"))
+    fire-rules
+    get-trace)
+````
```
Comment (Zylox) [inline _docs/listeners.md]: seems that the interpreter eats it anyway, but i dont think you need 4 here.

Comment (WilliamParker) [reply _docs/listeners.md]: Good catch @Zylox - I changed this to add Clojure highlighting with https://github.com/cerner/clara-site/pull/29/commits/1ba4001497ed9c903c2c0e35d83ee9148cacf6a2

With these changes there aren't any backticks anymore to create the code block.  This syntax-highlighting looks to be used in other places in the project as well.

**Pull request merged**